bblocks/import_tools/imf.py

```diff
@@ -0,0 +1,175 @@
+from bblocks.cleaning_tools.clean import clean_numeric_series
+from bblocks.import_tools.common import ImportData
+
+
+def _parse_sdr_links(url: str, concat_url: str) -> dict:
+    """Function to parse SDR tables from html
+
+    When called it finds the SDR tables and finds a relevant
+    date and link
+
+    Args:
+        url (str): URL to parse
+        concat_url: base url onto which a href is attached
+
+    Returns:
+        A dictionary with dates as keys and URLs as values
+    """
+
+    try:
+        content = requests.get(url).content
+    except ConnectionError:
+        raise ConnectionError(f'Could not read page: {url}')
+
+    soup = BeautifulSoup(content, 'html.parser')
+    table = soup.find_all('table')[4]
+    links = table.find_all('a')
+
+    return {link.string: f'{concat_url}{link.get("href")}' for link in links}
+
+
+def _get_tsv_url(url: str) -> str:
+    """retrieves the URL for the SDR TSV file
+
+    Args:
+        url (str): URL to parse
+
+    Returns:
+        A link to the TSV file
+    """
+
+    try:
+        content = requests.get(url).content
+    except ConnectionError:
+        raise ConnectionError(f'could not read page: {url}')
+    soup = BeautifulSoup(content, "html.parser")
+    href = soup.find_all("a", string="TSV")[0].get("href")
+
+    return f"https://www.imf.org/external/np/fin/tad/{href}"
+
+
+def _read_sdr_tsv(url: str) -> pd.DataFrame:
+    """Reads a TSV file with SDR data from the web to a pandas dataframe
+
+    Args:
+        url (str): link to the TSV file
+
+    Returns:
+        A pandas dataframe with SDR data
+    """
+
+    df = pd.read_csv(url, delimiter="/t", engine="python").loc[3:]
+    df = df["SDR Allocations and Holdings"].str.split("\t", expand=True)
+    df.columns = ["member", "holdings", "allocations"]
+
+    return (df.melt(id_vars='member', value_vars=["holdings", "allocations"])
+            .pipe(clean_numeric_series, series_columns="value")
+            .rename(columns={"variable": "indicator"})
+            .reset_index(drop=True)
+            )
+
+
+def _check_indicators(indicators: str | list) -> list:
+    """ """
+
+    accepted_indicators = ['allocations', 'holdings']
+    if isinstance(indicators, str):
+        if indicators not in accepted_indicators:
+            raise ValueError(f"{indicators} is not a valid indicator")
+        return [indicators]
+    elif isinstance(indicators, list):
+        for indicator in indicators:
+            if indicator not in accepted_indicators:
+                raise ValueError(f"{indicator} is not a valid indicator")
+        return indicators
+    elif indicators is None:
+        return ['allocations', 'holdings']
+
+
+class SDR(ImportData):
+    """ """
+
+    indicators: list = field(default_factory=list)
+
+    def load_indicator(self, indicators: Optional | str = None) -> ImportData:
+        """ """
+
+        # make sure indicators are valid
+        indicator_list = _check_indicators(indicators)
```

Comment:
This isn't totally coherent. Your `_check_indicators()` function accepts str and list, but it can only really ever get strings. Part of the logic in the function deals with accepting a list, but that will never happen. 

The name of this variable is also not 'right'. Only one indicator gets loaded at a time, so `indicator_list` will only ever contain a single string

Reply:
Thanks for catching that @jm-rivera. In my mind I thought to include a list if a user were to pass ['holdings', 'allocations'] but it is better to only allow 1 indicator to be passed at a time, or None to load all.